scripts/releasing/patch-trigger.js

```diff
@@ -0,0 +1,221 @@
+
+  const testMode = argv.test || process.env.TEST_MODE === 'true';
+
+  // Initialize GitHub API client only if not in test mode
+  let github;
+  if (!testMode) {
+    const { Octokit } = await import('@octokit/rest');
+    github = new Octokit({
+      auth: process.env.GITHUB_TOKEN,
+    });
+  }
+
+  const context = {
+    eventName: process.env.GITHUB_EVENT_NAME || 'pull_request',
+    repo: {
+      owner: process.env.GITHUB_REPOSITORY_OWNER || 'google-gemini',
+      repo: process.env.GITHUB_REPOSITORY_NAME || 'gemini-cli',
+    },
+    payload: JSON.parse(process.env.GITHUB_EVENT_PAYLOAD || '{}'),
+  };
+
+  // Get inputs from CLI args or environment
+  const headRef = argv.headRef || process.env.HEAD_REF;
+  const body = argv.prBody || process.env.PR_BODY || '';
+  const isDryRun = argv.dryRun || body.includes('[DRY RUN]');
+
+  if (!headRef) {
+    throw new Error(
+      'head-ref is required. Use --head-ref or set HEAD_REF environment variable.',
+    );
+  }
+
+  console.log(`Processing patch trigger for branch: ${headRef}`);
+
+  // Extract base version and channel from hotfix branch name
+  // New format: hotfix/v0.5.3/preview/cherry-pick-abc -> v0.5.3 and preview
+  // Old format: hotfix/v0.5.3/cherry-pick-abc -> v0.5.3 and stable (default)
+  const parts = headRef.split('/');
+  const version = parts[1];
+  let channel = 'stable'; // default for old format
+
+  if (parts.length >= 4 && (parts[2] === 'stable' || parts[2] === 'preview')) {
+    // New format with explicit channel
+    channel = parts[2];
+  } else if (context.eventName === 'workflow_dispatch') {
+    // Manual dispatch, infer from version name
+    channel = version.includes('preview') ? 'preview' : 'stable';
+  }
+
+  // Validate channel
+  if (channel !== 'stable' && channel !== 'preview') {
+    throw new Error(
+      `Invalid channel: ${channel}. Must be 'stable' or 'preview'.`,
+    );
+  }
+
+  const releaseRef = `release/${version}`;
+  const workflowId =
+    context.eventName === 'pull_request'
+      ? 'release-patch-3-release.yml'
+      : process.env.WORKFLOW_ID || 'release-patch-3-release.yml';
+
+  console.log(`Detected channel: ${channel}, version: ${version}`);
+  console.log(`Release ref: ${releaseRef}`);
+  console.log(`Workflow ID: ${workflowId}`);
+  console.log(`Dry run: ${isDryRun}`);
+
+  if (testMode) {
+    console.log('\n🧪 TEST MODE - No API calls will be made');
+    console.log('\n📋 Parsed Results:');
+    console.log(`  - Branch: ${headRef}`);
+    console.log(
+      `  - Channel: ${channel} → npm tag: ${channel === 'stable' ? 'latest' : 'preview'}`,
+    );
+    console.log(`  - Version: ${version}`);
+    console.log(`  - Release ref: ${releaseRef}`);
+    console.log(`  - Workflow: ${workflowId}`);
+    console.log(`  - Dry run: ${isDryRun}`);
+    console.log('\n✅ Channel detection logic working correctly!');
+    return;
+  }
+
+  // Try to find the original PR that requested this patch
+  let originalPr = null;
```

Comment:
Couldn't this cause silent failures if for some reason we didn't find a pr number